### ISSUE TITLE
WiFi Connect and bluetooth do not work together

### ISSUE THREAD

**Issue by vvkhazan** [balena-io/wifi-connect#271]
On balenaFins and similar devices, wifi and bluetooth are on the same chip. There is a known issue where using bluetooth (scanning for devices, pairing devices, etc.) will break wifi:

https://github.com/raspberrypi/linux/issues/1444
https://askubuntu.com/questions/1060094/bluetooth-audio-interfering-with-wifi

I believe this is only relevant for 2.4GHz wifi networks since bluetooth is also broadcast on that frequency. 5Ghz networks seem to work okay.

The access portal that WiFi Connect broadcasts is 2.4GHz. If bluetooth scans are happening when the access portal is up, it will not work properly.

Possible solution: add a command line option that allows the AP to be broadcast at 5GHz.

**Comment by majorz** [comment]
I was not aware of the existence of this kind of problem up until now. We were able to reproduce it indeed with a simple Bluetooth discovery in a loop at the same time the AP is running.

It should be possible to extend WiFi Connect to support running the AP in the 5GHz spectrum from what I can tell from the NetworkManager's documentation.

**Comment by majorz** [comment]
We did more testing on this and indeed RPi 3 B and  3 B+ are also affected by this.

We have to investigate further the issue on both the RPi models and balenaFin.

We will also investigate and implement dual-band AP support plus adding configurable options for choosing specific band.

**Comment by vvkhazan** [comment]
It's likely going to be a problem with any model that has WiFI and bluetooth on the same chip. Note that this also extends to connectivity in general -- if you're connected to a 2.4GHz WiFi network and doing bluetooth scans, the internet connectivity will likely be spotty and the scans will not be as accurate.

We've managed to solve this problem by using a USB bluetooth dongle and scanning using that instead of the built-in bluetooth chip.

**Comment by majorz** [comment]
Right, I would imagine there could be potential issues when sharing between WiFi and BT. Adding additional BT dongle is a nice idea.

**Comment by rosswesleyporter** [comment]
Customer also asked about this in https://jel.ly.fish/sales-thread-1-0-0-front-cnv-cq12znx
